cylc/flow/platforms.py

```diff
@@ -48,105 +46,66 @@
 #     Cylc9
 # remove at:
 #     Cylc9
-def get_platform(task_conf=None, task_id='unknown task', warn_only=False):
+def get_platform(
+    task_conf: Union[str, Dict[str, Any], None] = None,
+    task_id: str = 'unknown task'
+) -> Optional[Dict[str, Any]]:
     """Get a platform.
 
     Looking at a task config this method decides whether to get platform from
     name, or Cylc7 config items.
 
     Args:
-        task_conf (str, dict or dict-like such as OrderedDictWithDefaults):
-            If str this is assumed to be the platform name, otherwise this
-            should be a configuration for a task.
-        task_id (str):
-            Task identification string - help produce more helpful error
+        task_conf: If str this is assumed to be the platform name, otherwise
+            this should be a configuration for a task.
+        task_id: Task identification string - help produce more helpful error
             messages.
-        warn_only(bool):
-            If true, warnings about tasks requiring upgrade will be returned.
 
     Returns:
-        platform (platform, or string):
-            Actually it returns either get_platform() or
+        platform: Actually it returns either get_platform() or
```

Comment:
I don't like my original text - I think:

> platform: A platform definition object.

Because I think that the fuller explanation of the mechanics of platform selection are not required _here_.